clientd3d/graphics.c

```diff
@@ -330,13 +337,33 @@ void RedrawForce(void)
 	 Sleep(msSleep);
       }
    }
+
+   // Calcaute the FPS again aftter any adjustments from clamping.
+   fps = 1000 / (timeGetTime() - lastEndFrame);
+
+   // Update the last end frame and end the time period.
    lastEndFrame = endFrame;
    timeEndPeriod(1);
 
+   // Add the fps to the rolling window
+   frameTimes.push_back(fps);
+
+   // Remove the oldest fps if the window is full
+   if (frameTimes.size() > windowSize) {
+       frameTimes.erase(frameTimes.begin());
```

Comment:
Hm, this is going to happen every frame, even if showFPS is false, which is not great.  Use a std::deque instead?  Only do this if showFPS is on?